util/fork_func.c

```diff
@@ -0,0 +1,75 @@
+#include <sys/wait.h>
+
+#include <stdint.h>
+#include <unistd.h>
+
+#include "warnp.h"
+
+#include "fork_func.h"
+
+/**
+ * fork_func(func, cookie):
+ * Fork and run ${func} in a new process, with ${cookie} as the sole argument.
+ */
+pid_t
+fork_func(int (* func)(void *), void * cookie)
+{
+	pid_t pid;
+
+	/* Fork */
+	switch (pid = fork()) {
+	case -1:
+		warnp("fork");
```

Comment:
For consistency with the other `case`s, I would add `/* Error in fork system call. */`